trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/map-to-selected-payment-details-filters-view-model.ts

```diff
@@ -0,0 +1,29 @@
+import { PaymentDetailsFilters } from '@ukef/dtfs2-common';
+import { SelectedPaymentDetailsFiltersViewModel } from '../../../types/view-models';
+import { getPaymentDetailsTabHref } from './get-payment-details-tab-href';
+
+/**
+ * Map to selected payment details filters view model
+ * @param filters - The filter values
+ * @param reportId - The report id
+ * @returns the selected payment details filters view model
+ */
+export const mapToSelectedPaymentDetailsFiltersViewModel = (
+  filters: PaymentDetailsFilters,
+  reportId: string,
+): SelectedPaymentDetailsFiltersViewModel | null => {
+  if (!filters.facilityId && !filters.paymentCurrency && !filters.paymentReference) {
+    return null;
+  }
+  return {
```

Comment:
Finding the conditions/names a little hard to follow, what do you think about doing something like this instead?

```ts
  const mapped = {
    facilityId: null,
    paymentCurrency: null,
    paymentReference: null,
  };

  if (filters.facilityId) {
    mapped.facilityId = {
      value: filters.facilityId
      removeHref: getPaymentDetailsTabHref({ ...filters, facilityId: undefined }, reportId)
    };
  }

  if (filters.paymentCurrency) {
    // mapped.
  }

  return mapped;
```